PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/GetSyncPollerOnPollerFluxCheck.java

```diff
@@ -0,0 +1,185 @@
+         * @param expression Method call expression to visit
+         */
+        @Override
+        public void visitMethodCallExpression(@NotNull PsiMethodCallExpression expression) {
+            super.visitMethodCallExpression(expression);
+
+            // Check if the element is a method call expression
+            if (!(expression instanceof PsiMethodCallExpression)) {
+                return;
+            }
+
+            PsiMethodCallExpression methodCall = expression;
+
+            // Check if the method call is getSyncPoller
+            if (methodCall.getMethodExpression().getReferenceName().startsWith(METHOD_TO_CHECK)) {
+                boolean isAsyncContext = checkIfAsyncContext(methodCall);
+
+                if (isAsyncContext && isAzureClient(methodCall)) {
+                    holder.registerProblem(expression, ANTI_PATTERN_MESSAGE);
+                }
+            }
+        }
+
+        /**
+         * Helper method to check if the method call is within an async context.
+         * This method will check if the method call is on a PollerFlux type.
+         *
+         * @param methodCall Method call expression to check
+         * @return true if the method call is on a reactive type, false otherwise
+         */
+        private boolean checkIfAsyncContext(@NotNull PsiMethodCallExpression methodCall) {
+            PsiExpression expression = methodCall.getMethodExpression().getQualifierExpression();
+
+            // Check if the method call is on a reactive type
+            if (expression == null) {
+                return false;
+            }
+            PsiType type = expression.getType();
+
+            // Check if the type is a reactive type
+            if (type == null) {
+                return false;
+            }
+            String typeName = type.getCanonicalText();
+
+            // Check for PollerFlux type
+            if (typeName != null && typeName.contains("PollerFlux")) {
+                return true;
+            }
+            return false;
+        }
+
+        /**
+         * Helper method to check if the method call is on an Azure SDK client.
+         * This method will check if the method call is on a class that is part of the Azure SDK.
+         * @param methodCall Method call expression to check
+         * @return true if the method call is on an Azure SDK client, false otherwise
+         */
+        private boolean isAzureClient (@NotNull PsiMethodCallExpression methodCall){
+
+            PsiClass containingClass = PsiTreeUtil.getParentOfType(methodCall, PsiClass.class);
+
+            // Check if the method call is on a class
+            if (containingClass == null) {
+                return false;
+            }
+            String className = containingClass.getQualifiedName();
+
+            // Check if the class is part of the Azure SDK
+            if (className != null && className.startsWith("com.azure.")) {
```

Comment:
Lets put `com.azure`  in a common util, actually we could even consider taking this method into a Util so that it can be used by all rules.

Reply:
I have the ruleConfig.AZURE_PACKAGE_NAME common util :)